system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -28,8 +28,10 @@ this.TopStoriesFeed = class TopStoriesFeed {
       const prefs = new Prefs();
       const options = JSON.parse(prefs.get("feeds.section.topstories.options"));
       const apiKey = this._getApiKeyFromPref(options.api_key_pref);
-      this.stories_endpoint = this._produceUrlWithApiKey(options.stories_endpoint, apiKey);
-      this.topics_endpoint = this._produceUrlWithApiKey(options.topics_endpoint, apiKey);
+      const locale = Services.locale.getRequestedLocale();
+      this.stories_endpoint = this._produceFinalEndpointUrl(options.stories_endpoint, apiKey, locale);
```

Comment:
We don't need to fix the "later" than init network now, but I was thinking not that much long after init. It's just that anything done synchronously during the constructor and INIT counts against one of the talos performance metrics.